src/Distribution/AutoUpdate.hs

```diff
@@ -53,6 +53,18 @@ import Distribution.AutoUpdate.VerifySignatures
 import qualified StoryMode.Client
 import qualified StoryMode.AutoUpdate
 
+-- I know there's orphan instances here... upgrade GHC if you don't want these!
+-- They are necessary for a later use of fmapM_
+#if MIN_VERSION_base(4,7,0)
+#else
+-- Taken directly from the source of base version 4.8.0.0
+instance Foldable (Either a) where
+    foldMap _ (Left _) = mempty
+    foldMap f (Right y) = f y
+
+    foldr _ z (Left _) = z
+    foldr f z (Right y) = f y z
+#endif
```

Comment:
There's a package called `base-compat` that tries to provide a compatibility layer on top of different versions of `base`. Maybe we could use that. We would get rid of `CPP` then.